packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -1039,6 +1045,43 @@ export const GridMixin = (superClass) =>
       }
     }
 
+    /** @private */
+    __isTargetInViewport(target) {
+      const targetRect = target.getBoundingClientRect();
+      const gridRect = this.getBoundingClientRect();
+
+      if (targetRect.left < gridRect.left || targetRect.right > gridRect.right) {
```

Comment:
Could refactor to only check the bounds once, something like:
```js
__isCellFullyVisible(cell) {
  if (cell.hasAttribute('frozen') || cell.hasAttribute('first-frozen-to-end')) {
    // Forzen cells are always fully visible
    return true;
  }

  let { left, right } = this.getBoundingClientRect();

  const frozen = [...cell.parentNode.children].find((cell) => cell.hasAttribute('last-frozen'));
  if (frozen) {
    const frozenRect = frozen.getBoundingClientRect();
    left = this.__isRTL ? left : frozenRect.right;
    right = this.__isRTL ? frozenRect.left : right;
  }

  const frozenToEnd = [...cell.parentNode.children].find((cell) => cell.hasAttribute('first-frozen-to-end'));
  if (frozenToEnd) {
    const frozenToEndRect = frozenToEnd.getBoundingClientRect();
    left = this.__isRTL ? frozenToEndRect.right : left;
    right = this.__isRTL ? right : frozenToEndRect.left;
  }
  const cellRect = cell.getBoundingClientRect();
  return cellRect.left >= left && cellRect.right <= right;
}
```

Reply:
Thanks, updated.